src/line.js

```diff
@@ -0,0 +1,134 @@
+import * as d3 from "d3"
+
+/**
+ * Render all bars on the canvas div.
+ * @param {Spotfire.DataView} dataView
+ */
+
+export async function line(dataView) {
+
+    d3.selectAll('path').remove();
+
+    console.log(dataView)
+
+
+
+    const allval = await dataView.allRows();
+    // const pp = dataView;
+
+    // console.log(pp)
+    console.log(allval)
+    let cval = [];
+    // console.log(allval);
+    for (let i of allval) {
+        const val = Object.values(i);
+        cval.push(val[0][2]);
+        cval.sort((a, b) => { return b - a })
+    }
+    console.log(cval);
+
+    // for (let i of cval) {
+    //     finalval.push(i[2])
+    //     finalval.sort((a, b) => { return a - b })
+    // }
+    // console.log(pp)
+    let j = 0;
+    for (let i of await dataView.allRows()) {
+        Object.values(i)[0][2] = cval[j]
+        j++
+    }
+
+
+    const cumal = cval.reduce((pre, curr) => { return pre + curr })
+
+    console.log(cumal)
+    function calpercantges(cval, cumal) {
+        let percentages = [];
+        for (let i of cval) {
+            percentages.push(parseFloat(((i * 100) / cumal).toFixed(2)))
+        }
+        return percentages;
+    };
+
+    const final_percantag = calpercantges(cval, cumal);
+
+    function calcValOfPercantages(cval, final_percantag) {
+        let val = [];
+        let secval = [0]; //this name has to change 
+        for (let i of final_percantag) {
+            val.push(parseFloat(((i * cval[0]) / 100).toFixed(2)))
+        }
+        for (let i = 0; i < val.length; i++) {
+            secval.push(parseFloat((val[i] + secval[i]).toFixed(2)))
+        }
+        console.log(val)
+        secval.shift();
+        return secval;
+    };
+    const fi = calcValOfPercantages(cval, final_percantag)
+    const maxValueHeight = Math.max(...fi);
+    const maxValueWidth = fi.length;
+
+
+
+    function createDataSet(a) {
+        const last = [];
+        for (let i = 0; i < a.length; i++) {
+            const ar = [];
+            ar.push(i + 1)
+                ; ar.push(a[i]);
+            last.push(ar);
+        };
+        return last
+    };
+    const h = createDataSet(fi)
+    console.log(h);
+
+
+    //====================================================================//
+    var dataset1 = h;
```

Comment:
It's unnecessary to assign h to dataset1 (I assume this is a leftover from when testing with mock-up data)

Reply:
The code has been refactored and the comments have been applied